### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by JonMaxim**
This is a work in progress. Feedback is welcome and appreciated.

### CODE REVIEW

**Comment by wbclark** [comment]
We're planning to add further examples of external PostgreSQL database with encrypted connection, and using the Pulp 2 migration plugin.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,22 @@
...
-Puppet module for setting up Pulp 3 as part of Katello installation
+This is a Puppet module for setting up Pulp 3 as part of a Katello installation. It can also be utilized to deploy a standalone Pulp 3.
+
+## EXAMPLES
+
+### Basic Standalone Deployment
+
+Using the module to manage both the database and Apache Virtual Host configuration.
+
+```
```
Comment (ekohl) [inline README.md]: You can get syntax highlighting
```suggestion
```puppet
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,64 @@
...
+  postgresql_db_password    => 'verysecurepassword',
+  postgresql_db_ssl         => true,
+  postgresql_db_ssl_require => true,
+  postgresql_db_ssl_cert    => '/etc/pki/katello/certs/pulpcore-database.crt',
+  postgresql_db_ssl_key     => '/etc/pki/katello/private/pulpcore-database.key',
+  postgresql_db_ssl_root_ca => '/etc/pki/tls/certs/ca-bundle.crt',
+}
+```
+
+## CONTRIBUTING
```
Comment (ekohl) [inline README.md]: There's already https://github.com/theforeman/puppet-pulpcore/blob/master/CONTRIBUTING.md

**Comment by JonMaxim** [comment]
> The README.md file was made executable

Thanks @ekohl. I have fixed the permissions issue and will work on tagged examples soon. Appreciate the feedback.

**Comment by JonMaxim** [comment]
Added an updated `README.md` and also moved examples to `init.pp`.

**Comment by ekohl** [comment]
You can also use `bundle exec rake strings:generate:reference` to generate the `REFERENCE.md`. Do not commit it, but it's what we do on release.

**Comment by JonMaxim** [comment]
I submitted updates to the `README.md` and `init.pp`.

@ekohl I still did not find examples populating in `REFERENCE.md` after I successfully executed `bundle install` then `bundle exec rake strings:generate:reference`. Do you have any ideas of other things to try? Thank you!

File: manifests/init.pp [hunk 3]
```diff
@@ -1,5 +1,46 @@
 # Manage your next generation Pulp server
 #
+# @example Basic Standalone Deployment
+# include pulpcore
```
Comment (ekohl) [inline manifests/init.pp]: You need to indent after using such a tag
```suggestion
#   include pulpcore
```

File: manifests/init.pp [hunk 4]
```diff
@@ -80,9 +121,7 @@
 #
 # @param remote_user_environ_name
 #   Django remote user environment variable
-#
-# @example
-#   include pulpcore
+
```
Comment (ekohl) [inline manifests/init.pp]: This empty line breaks the documentation tooling

**Comment by ekohl** [comment]
Now that 0.2.0 is released there's an example: https://forge.puppet.com/theforeman/pulpcore/reference

**Comment by ekohl** [comment]
Closing due to inactivity.

**Pull request closed without merging**